### PR TITLE
MLIBZ-1542: Push Notifications

### PR DESCRIPTION
**Pull request by thomasconner**
#### Description
This PR fixes several issues with push notifications not working on Titanium.

#### Changes
- Update core racks with Titanium middleware.
- Fix push notifications issues.
- Update configs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,26 +36,25 @@
...
   "dependencies": {
-    "es6-error": "^3.2.0",
-    "es6-promise": "^4.0.4",
-    "kinvey-node-sdk": "3.3.5",
-    "lodash": "^4.16.2",
-    "parse-headers": "^2.0.1",
-    "rxjs": "5.0.0-beta.12",
-    "webpack-external-require": "1.0.0"
+    "es6-promise": "4.0.5",
+    "kinvey-node-sdk": "file:///Users/Thomas/Documents/Kinvey/Development/SDKs/JavaScript/Node/SDK",
```
Comment (tejasranade) [inline package.json]: @thomasconner this needs cleanup.

File: package.json [hunk 2]
```diff
@@ -36,26 +36,25 @@
     "version": "npm run build && git add -A dist"
   },
   "dependencies": {
-    "es6-error": "^3.2.0",
-    "es6-promise": "^4.0.4",
-    "kinvey-node-sdk": "3.3.5",
-    "lodash": "^4.16.2",
-    "parse-headers": "^2.0.1",
-    "rxjs": "5.0.0-beta.12",
```
Comment (tejasranade) [inline package.json:44]: @thomasconner I see rxjs in the `webpack.config` but removed from the `package.json`. Wouldn't this affect developers that use npm with this SDK?

**Review by tejasranade — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (thomasconner) [reply package.json:44]: RxJS is a dependency of the core and not Titanium SDK. The reference you see in webpack.config.js is a local file to fix the issue with getting the root. See https://github.com/Kinvey/titanium-sdk/blob/ef8f07d77bc212d8da7d6b277caf887a96189886/src/rxjs.js.

File: package.json (hunk 1 above)
Comment (thomasconner) [reply package.json]: Fixed

**Review by tejasranade — APPROVED** [review]

**Pull request merged**